blackbox/blackbox/reporter.go

```diff
@@ -0,0 +1,50 @@
+package blackbox
+
+import (
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+// Result represent result from scrape operation
+type Result struct {
+	responses []http.Response
+	jobName   string
+	err       error
+}
+
+// logResult reports results into zap logger.
+type logResult struct {
+	l *zap.Logger
+}
+
+func NewLogResult() (*logResult, error) {
+	devLog, err := zap.NewDevelopment()
+	return &logResult{
+		l: devLog,
+	}, err
+}
+
+func (l *logResult) Report(resp <-chan Result) {
```

Comment:
if in the end - reporting is only required with the log and not something else - there is no need to gather it - log is concurrency free structure so it can be executed in multiple go routines at the same time